Remove temporary switch 'dwarfeh'

mathias-lang-sociomantic opened the pull request:
It was introduced in P.R. #5305 to help debugging dwarfeh,
however it was supposed to be a temporary switch, but made it into
a release.  Some tests that started to depend on it where also amended.
Luckily it was never documented so we just remove it.

@WalterBright : This isn't a revert of your P.R. since there are two changes ([This looks like a bugfix](https://github.com/D-Programming-Language/dmd/pull/5305/files#diff-eb9cdf480cc83e486a88311d1df5d141R1554) and [this isn't 100% related to the switch](https://github.com/D-Programming-Language/dmd/pull/5305/files#diff-5c4ac22992d1aaf56043c3b55a55c39eR113)) which seemed unrelated to the switch itself

mathias-lang-sociomantic committed:
Remove temporary switch 'dwarfeh'

It was introduced in P.R. #5305 to help debugging dwarfeh,
however it was supposed to be a temporary switch, but made it into
a release.  Some tests that started to depend on it where also amended.
Luckily it was never documented so we just remove it.

braddr commented:
There's at least one open regression bug filed due to the switch to dwarf eh.  I don't think a release or two of allowing a command-line switch to exist to fall back to the old way is a problem, rather actually a good idea.

mathias-lang-sociomantic commented:
I agree, however DWARF EH is enabled by default with no way to disable it. This switch has no effect on what happens as you can see in [the backend code](https://github.com/D-Programming-Language/dmd/blob/1a211521764c971bcd3c892cef159f7df2e90f9f/src/backend/backconfig.c#L97). It is just ignored.

braddr commented:
Yup.. I didn't look deeper, sorry.  I'd be more in favor of fixing it so the switch worked than removing it.

mathias-lang-sociomantic commented:
Do we want to continue to support `DM_EH` ? It looks like we're on the way to drop it, as only Win64, OpenBSD and Solaris are still using it.

Also the purpose of the switch was to enable DWARF conditionally during debugging. I guess @WalterBright had good reasons not to provide an user-facing mean to switch back to `DM_EH` once Dwarf support was in ?

WalterBright commented:
After https://github.com/D-Programming-Language/dmd/pull/5510 is pulled, I'm fine with pulling this one. `DM_EH` will continue as Win64 uses it, and we may need it to support some random platform that doesn't have Dwarf.

WalterBright commented:
> There's at least one open regression bug filed due to the switch to dwarf eh.

I presume you mean https://issues.dlang.org/show_bug.cgi?id=15779, but that isn't about the `-dwarfeh` compiler switch.

> I don't think a release or two of allowing a command-line switch to exist to fall back to the old way is a problem, rather actually a good idea.

I don't see a further point to the switch, and any code that decided to throw an undocumented switch is not supported.

braddr commented:
That's the bug, and it's not directly about the switch, but if the switch worked, then the regression would have an easy work around until fixed.  Sadly, it doesn't and so there isn't.

Geod24 commented:
Well there's an easy workaround for this one: Remove the `-gx` switch.
Also, there might be some other bug in `-gx`, as @marenz mentioned he had two `ref bool` parameter being swapped.

WalterBright commented:
> but if the switch worked, then the regression would have an easy work around until fixed.

The switch enables dwarf eh, it does not disable it, so it would be of no use for the regression. But a simpler workaround is available - don't use `-gx`.

mathias-lang-sociomantic commented:
So it fails under archs which don't support DWARFEH because it produces a different error message.
How do you feel about excluding those from the test at the moment (printing a message) ?

andralex commented:
Auto-merge toggled on

braddr commented:
Auto-merge toggled off

braddr commented:
Until it actually passes on all platforms, no reason to be giving it higher priority build status.

mathias-lang-sociomantic commented:
Blocked on https://github.com/D-Programming-Language/dmd/pull/5510#issuecomment-196453739

mathias-lang-sociomantic committed:
Allow to catch C++ exceptions on OSX32 as well

Now that DWARFEH is enabled and works on OSX32 we can enable catching C++ exceptions.

WalterBright commented:
Need to fix the Darwin_64_32 target, seems to be a checkwhitespace problem. Otherwise looks ready.

Geod24 commented:
@WalterBright See my comment [here](https://github.com/D-Programming-Language/dmd/pull/5510#issuecomment-197621426).
Dwarf EH wasn't enabled by default on this platform, and when I tried to, it resulted in a segfault in checkwhitespace. On [the last run](https://auto-tester.puremagic.com/show-run.ghtml?projectid=1&runid=2002670&isPull=true) it seemed to be a bus error. Nevertheless, I don't see how that could be caused by removing `-dwarfeh`, so I assume it's a problem in the implementation on OSX32.

WalterBright commented:
I thought it was enabled. Looks like I have more work to do.

mathias-lang-sociomantic commented:
Rebased on master / fixed / green.

WalterBright commented:
Auto-merge toggled on
